### PR TITLE
feat: Relates to #264. SUI-based drop-down menu with Backup Account and Add Tokens

### PR DESCRIPTION
**Pull request by ltfschoen**
* **Usage**: On the "Account List" page, click and account to be taken to its "Account" page.
Then click the icon with the three dots to open the menu. Or click anywhere else to hide it.

* Replaces PR #318.

* Previous review comments adopted for the menu:
  * Account page menu now based on Semantic UI (SUI) Popup instead of custom job.
    * Reference: https://react.semantic-ui.com/modules/popup/
  * Place the modal above the button (this means we cannot use popup arrow)
  * Max menu height
  * Transparent background around the menu (i.e. not grey like a modal)

* Proposed changes to previous discussion:
  * Use a SUIButton style (i.e. black border, white background)
  for menu items instead of just ordinary text
  * If multiple menu items have short names they will be next to each other
  potentially saving the user from scrolling too much if we end up having many items

**Screenshots**

How it looks with just the two current menu items:
screen shot 2019-01-13 at 12 47 50 pm

How it would look if we had lots of menu items:
screen shot 2019-01-13 at 12 46 38 pm

View of the page with just the "three dots" menu icon in the top right that a user can click on:
screen shot 2019-01-13 at 12 48 10 pm

### CODE REVIEW

**Comment by Tbaut** [comment]
Thanks Luke. I haven't tested it yet. Can we make it so that it looks like on the mockup (#264)? With simple lines instead of buttons.

**Comment by ltfschoen** [comment]
> Thanks Luke. I haven't tested it yet. Can we make it so that it looks like on the mockup (#264)? With simple lines instead of buttons.

Sure will do

**Comment by ltfschoen** [comment]
@Tbaut I've updated it so the menu items are like before but with slightly darkened background colour on hover.

Screenshot with two items:
screen shot 2019-01-13 at 11 44 11 pm

Screenshot with multiple items causing scrollbar to appear:
screen shot 2019-01-13 at 11 45 27 pm

**Comment by Tbaut** [comment]
Just tested, that looks pretty good! A couple comments:
- Can we have this behavior that the menu stays displayed as long as the user doesn't click away? (no autoclose)
- Could we use the same styling as the other menus for hovers (black on dark grey). Like 
image

**Comment by ltfschoen** [comment]
> Just tested, that looks pretty good! A couple comments:
> 
> * Can we have this behavior that the menu stays displayed as long as the user doesn't click away? (no autoclose)
> * Could we use the same styling as the other menus for hovers (black on dark grey). Like
>   image

@Tbaut I've pushed commits to address your review comments.

**Comment by Tbaut** [comment]
- Misunderstanding regarding the 3 dots sorry. I'm totally fine with them.
I was talking about the **styling** of the menu items on hover (the list "backup account", "add token"). On mouse hover, instead of having the inverted colors + the animation, to simply have the same styling as what we have on other menus, without animation to keep things consistent e.g 
image

```
.header-nav a:hover, .header-nav a:link:hover, .header-nav a:visited:hover {
    background: rgba(221, 221, 221, 0.5);
    color: #222;
}
```

- Also regarding the click away, one thing missing, similarly to https://github.com/paritytech/fether/pull/318#issuecomment-449424025 it would be great if the click wouldn't trigger anything (there should be a layer between the token layer and the menu) so that if you click on an token card to discard the menu, you don't end up on the sendTx screen.

**Comment by ltfschoen** [comment]
@Tbaut Sorry about that. I've pushed commits to address those comments (i.e. layer under menu to prevent clicking another link upon menu blur click, restore three dot icon and fix styling of menu including hover for consistency)

**Comment by Tbaut** [comment]
- I think it works well and looks good :tada: 
- Given the fact that the big PR (Signer) will be merged soon and they conflict (do not show a specific menu item for Signer account) I suggest that we wait to merge it right after, is that ok?

**Comment by ltfschoen** [comment]
> * I think it works well and looks good 🎉
> * Given the fact that the big PR (Signer) will be merged soon and they conflict (do not show a specific menu item for Signer account) I suggest that we wait and merge this one right after the big one, is that ok?

Yes sure, Signer is priority

**Comment by Tbaut** [comment]
That was quick, the big PR got merged, can you check to solve the conflicts @ltfschoen ?

**Comment by ltfschoen** [comment]
> That was quick, the big PR got merged, can you check to solve the conflicts @ltfschoen ?

Yes, having a look now

**Comment by ltfschoen** [comment]
@Tbaut I've pushed a commit that merges latest master and fixes Signer merge conflicts

**Comment by Tbaut** [comment]
Top, works well :+1:

File: packages/fether-react/src/assets/sass/components/_header-nav.scss [hunk 1]
```diff
@@ -84,6 +86,10 @@
       &.-new {
         background-image: url('./assets/img/icons/plus.svg');
       }
+
+      &.-menu {
+        background-image: url('./assets/img/icons/menu.svg');
```
[the remark this answers is not in the document]
Comment (ltfschoen) [reply packages/fether-react/src/assets/sass/components/_header-nav.scss]: I haven't had any luck with this, and it's my second attempt.

I've tried and tried to get semantic-ui icons to work, but can't get an icon to appear. It just shows:

screen shot 2019-01-18 at 8 39 19 am

I've used the following code:

paritytech/fether/packages/fether-react/src/Tokens/Tokens.js
```
<Icon
  className='ellipsis horizontal'
  color='red'
  fitted={true}
  name='ellipsis horizontal'
  onClick={this.handleToggleMenu}
  ref={this.handleMenuRef}
  size='massive'
/>
```

paritytech/fether/packages/fether-react/src/assets/sass/components/_header-nav.scss
```
// .Pulse(@duration) {
//   -webkit-animation-duration: @duration;
//   animation-duration: @duration;
//   -webkit-animation-name: pulse;
//   animation-name: pulse;
// };

i {
  &.icon {
    font-size: 2.5rem !important;    
    width: 2rem !important;      
    text-shadow: 0 1px #ddd !important;
    cursor: pointer !important;
    padding: .5em !important;
    margin: .5em 0 .5em 0 !important;
    position: relative !important;
    text-align: center !important;
    color: green;
    background: blue;

    &:hover {
      color: #c00 !important;
      // .Pulse(500ms) !important;
    }             
  }
}
```

I've tried using different icon names i.e. blind, wechat.

I've tried adding all the latest dependencies to fether-react/package.json
```
"semantic-ui": "^2.4.2",
"semantic-ui-css": "^2.4.1",
"semantic-ui-react": "^0.84.0",
"semantic-ui-icon": "^2.3.3"
```

I've tried just using an older version of semantic-ui
```
"semantic-ui": "^2.3.2"
```

I've tried just using semantic-ui-icon by itself

```
"semantic-ui-icon": "^2.3.3"
```

I've tried using an old CDN version instead and removing all the semantic-ui dependencies from package.json in fether-react, using this as a reference: https://codepen.io/IronGeek/pen/mezyey

paritytech/fether/packages/fether-react/public/index.html
```
<link href="//oss.maxcdn.com/semantic-ui/2.1.6/semantic.min.css" rel="stylesheet" />
<script src="http://oss.maxcdn.com/semantic-ui/2.1.6/semantic.min.js"></script>
```

[the remark this answers is not in the document]
Comment (ltfschoen) [reply packages/fether-react/src/assets/sass/components/_header-nav.scss]: I'm going to park this and come back to it, perhaps even raise an issue at https://github.com/Semantic-Org/Semantic-UI/issues

**Comment by ltfschoen** [comment]
@amaurymartiny I've pushed changes to address your review comments. I haven't had any success implementing Icon from semantic-ui. I've shared what I've tried but I could do with some help.

File: packages/fether-react/src/Tokens/Tokens.js [hunk 2]
```diff
@@ -14,12 +14,37 @@ import withAccount from '../utils/withAccount';
...
+
+    let menuItems = [
+      {
+        name: 'Add Tokens',
+        onClick: () => history.push(`/whitelist/${address}`)
+      }
+    ];
+
+    if (this.isParitySignerAccount() === false) {
+      menuItems = [backupAccountItem, ...menuItems];
```
Comment (amaurymartiny) [inline packages/fether-react/src/Tokens/Tokens.js]: You can use unshift() here

File: packages/fether-react/src/Tokens/Tokens.js [hunk 3]
```diff
@@ -14,12 +14,37 @@ import withAccount from '../utils/withAccount';
...
+      account: { address },
+      history
+    } = this.props;
+
+    const backupAccountItem = {
+      name: 'Backup Account',
+      onClick: () => history.push(`/backup/${address}`)
+    };
+
+    let menuItems = [
```
Comment (amaurymartiny) [inline packages/fether-react/src/Tokens/Tokens.js]: see above comment, this can be const

File: packages/fether-ui/src/MenuPopup/MenuPopup.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export const MenuPopup = ({ menuItems, ...otherProps }) => (
+  <SUIPopup {...otherProps}>
+    <div className='popup-screen'>
+      <SUIPopup.Content>
+        {menuItems &&
+          menuItems.map(item => (
+            <div
+              className='popup-screen_item'
+              key={item.name}
+              onClick={() => item.onClick()}
```
Comment (amaurymartiny) [inline packages/fether-ui/src/MenuPopup/MenuPopup.js]: onClick={item.onClick}

File: packages/fether-react/src/assets/sass/components/_header-nav.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (amaurymartiny) [reply packages/fether-react/src/assets/sass/components/_header-nav.scss]: No problem, let's forget this for now, and maybe come back with #158.

**Comment by ltfschoen** [comment]
@amaurymartiny I've pushed commits addressing your latest comments. We can raise an issue to try and implement SUIIcon instead of using the svg's as you mentioned

File: packages/fether-ui/src/MenuPopup/MenuPopup.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+export const MenuPopup = ({ menuItems, ...otherProps }) => (
+  <SUIPopup {...otherProps}>
+    <div className='popup-screen'>
+      <SUIPopup.Content>
+        {menuItems &&
+          menuItems.map(item => (
+            <div
+              className='popup-screen_item'
+              key={item.name}
+              onClick={() => item.onClick}
```
Comment (amaurymartiny) [inline packages/fether-ui/src/MenuPopup/MenuPopup.js]: onClick={item.onClick}

**Comment by Tbaut** [comment]
The clickaway behavior seems to have come back again after the changes. I opened an issue: https://github.com/paritytech/fether/issues/383

**Pull request merged**